lib/active_merchant/billing/gateways/pay_trace.rb

```diff
@@ -232,16 +242,16 @@ def add_payment(post, payment)
       def add_level_3_data(post, options)
         post[:invoice_id] = options[:invoice_id] if options[:invoice_id]
         post[:customer_reference_id] = options[:customer_reference_id] if options[:customer_reference_id]
-        post[:tax_amount] = amount(options[:tax_amount]) if options[:tax_amount]
-        post[:national_tax_amount] = amount(options[:national_tax_amount]) if options[:national_tax_amount]
+        post[:tax_amount] = options[:tax_amount].to_i if options[:tax_amount]
```

Comment:
I have a similar worry here, that weird values passed in to these fields will have unintended results.
```
'test'.to_i
=> 0

''.to_i
=> 0
```
So I just want to ask if this is something we should guard against? Maybe it doesn't even matter because maybe a zero in these fields would have no impact.

Please excuse me if this is a scenario you have considered.

Reply:
Thanks for pointing this out, that's a great thing to consider. In this case, the value of `0` is fine. It would be accepted by the gateway and wouldn't have an impact on the transaction.